File: average.js
const readline = require('readline-sync');

function gradeAverage(subject1, subject2, subject3) {
  return (subject1 + subject2 + subject3) / 3;
}

(function table() {
  try {
    while (true) {
      let subj1_grade = readline.question("Type the subject 1 grade:\n");
      while (isNaN(parseFloat(subj1_grade))) {
        subj1_grade = readline.question("Type a numeric input:\n");
      }

      let subj2_grade = readline.question("Type the subject 2 grade:\n");
      while (isNaN(parseFloat(subj2_grade))) {
        subj2_grade = readline.question("Type a numeric input:\n");
      }

      let subj3_grade = readline.question("Type the subject 3 grade:\n");
      while (isNaN(parseFloat(subj3_grade))) {
        subj3_grade = readline.question("Type a numeric input:\n");
      }

      console.log(`The average of the grades is: ${gradeAverage(parseFloat(subj1_grade), parseFloat(subj2_grade), parseFloat(subj3_grade))}`);

      const exit = readline.question("To exit type y or press Enter to continue\n");
      if (exit === "y") {
        return;
      }
    }
  } catch (error) {
    console.log("\nInvalid input. Please, type a text.\n");
    table();
  }
})()

File: average.py
"""
    Created by Edwildson Coelho Rodrigues
    https://github.com/edwildson
    https://www.linkedin.com/in/edwildson/

    Grade Average
"""

def grade_average(subject1, subject2, subject3):
    return (subject1 + subject2 + subject3)/3

def table():
    try:
        while True:
            subj1_grade = input("Type the subject 1 grade:\n")
            if not float(subj1_grade):
                subj1_grade = input("Type a numeric input:\n")
            subj2_grade = input("Type the subject 2 grade:\n")
            if not float(subj2_grade):
                subj1_grade = input("Type a numeric input:\n")
            subj3_grade = input("Type the subject 3 grade:\n")
            if not float(subj3_grade):
                subj1_grade = input("Type a numeric input:\n")

            print(f"The average of the grades is: {grade_average(float(subj1_grade), float(subj2_grade), float(subj3_grade))}")


            exit = input("To exit type y or enter to continue\n")
            if exit == "y":
                return

    except ValueError:
        print("\nInvalid input. Please, type a text.\n")
        table()

if __name__ == "__main__":
    table()